### PR TITLE
Add some documentation, tests

### PR DESCRIPTION
**Pull request by jamesob**
and there's a little bit of refactoring in here too.

I'm not entirely sure why the compiled `demo.map` changed - my refactorings shouldn't have changed anything other than possible ordering differences among equally common default ASN candidates due to use of a `Counter` (vs. maxing a dict). I can dig into this if you think it's important, or simply revert the refactoring altogether.

Consider this my tested ACK. :)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,3 +9,6 @@ demo.dat.xz: ipv4.dump ipv6.dump birdparse.py
 
 demo.map: demo.dat.xz buildmap.py
 	xz -d <demo.dat.xz | python3 buildmap.py >demo.map
+
+test: demo.map
+	python3 testmap.py
```
Comment (dongcarl) [inline Makefile]: I believe you also need to add
```
.PHONY: clean
```

**Comment by naumenkogs** [comment]
Tested ACK.
I reviewed all the comments, they seem correct, as well as the new test.
This PR makes the library friendlier to whoever will be looking at the code (including us in the future).
@sipa do you wanna take a look?

Also, no idea why the output is different size now, since everything seems deterministic :)

**Comment by jonatack** [comment]
Concept ACK -- great stuff @jamesob

**Comment by sipa** [comment]
going to merge this, but i don't know if this python code will survive much longer (better gathering-dumps logic is in asmap-rs, and better compilation to asmap format is in the Bitcoin Core PR)

**Pull request merged**